<style>A {	TEXT-TRANSFORM: none; TEXT-DECORATION: none } A:hover { COLOR: red; TEXT-DECORATION: underline } A.s { TEXT-TRANSFORM: none; TEXT-DECORATION: none;	color:black; } A.s:hover {	COLOR: white; TEXT-DECORATION: none; } A.s0 { TEXT-TRANSFORM: none; TEXT-DECORATION: none;  color:#FFFF99;	font-weight:bold; } A.s0:hover { TEXT-DECORATION: underline; color:#FFFF99;	font-weight:bold; } A.t { TEXT-DECORATION: none; color:#FFFF66; } A.t:hover {	color:#FFFF66; text-decoration: underline; } A.x { TEXT-DECORATION: none; color:white; font-weight:bold;} A.x:hover {	color:#FFFF66; text-decoration: underline; font-weight:bold;} A.bt { padding:2px 3px 2px 3px; margin: 0px; height: 20px; color:blue; font-family:Verdana, Arial, Helvetica, sans-serif; font-size:12px;} A.bt:hover {background-color:#E6E6E6; padding:1px 2px; margin: 0px; border: 1px solid #999; height: 20px; color:#000066; text-decoration: none; cursor: hand;} div.sd1 { position:absolute; left:0px} div.sd2 { position:absolute; right:0px} div.ptt { text-align:center; font-family:Arial,Helvetica,sans-serif; font-size:18pt; font-weight:bold; color:blue;} div.plm { text-align:center; font-family:Arial,Helvetica,sans-serif; font-size:12pt;} div.ptx { font-family:"Times New Roman",Times,serif; font-size:12pt;} p.pst { text-align:left; font-family:Arial,Helvetica,sans-serif; font-size:18pt; font-weight:bold; color:blue;} pre.sio { font-family:Courier New,Courier,monospace; font-size:12pt;} th.h {color:#FFFFFF; font-family:Arial, Helvetica, sans-serif; width:20%;} td.h { font-family:Arial, Helvetica, sans-serif; } th.in {color:#FFFFFF; text-align:center; background-color:#6589d1; border-color:#FFFFFF; font-family:Arial, Helvetica, sans-serif; font-weight:bold; } table.a {background-image:url('images/table_back.jpg'); border-collapse:collapse; font-family:"Times New Roman", Times, serif; } table.h1 {border-collapse:collapse; width:100%; background-color:#BECEF1; border-width:0px;} table.h2 {width:100%; background:url('images/bar.jpg') repeat-y; border-width:0px;} tr.in {color:#FFFFFF; text-align:center; background-color:#6589d1; font-family:Arial, Helvetica, sans-serif; font-weight:bold; } tr.h { background-color:#6589d1; vertical-align:top; color:#FFFFFF; font-family:Arial, Helvetica, sans-serif; } span.b1 {color:#6589D1; font-weight:bold; font-size:x-small } span.b2 {color:gray; font-size:x-small } span.n1 {color:red; font-size:x-small } ul.v {filter:Alpha(opacity=100); overflow:hidden; } table.survey {border-collapse:collapse; font-family:"Times New Roman", Times, serif;}
.stdft { font-family:"Times New Roman", Times, serif; } span.h_txt { color: blue; } span.h_award { color: red; }  span.h_nocontest { color: #1a5cc8; }  span.h_contest { color: red;} 
</style><div class="ptt" lang="en-US">Human Gene Functions</div>
<div class="plm"><table align="center"><tr><td><b>Time Limit:</b> 1000MS</td><td width="10px"></td><td><b>Memory Limit:</b> 10000K</td></tr><tr><td><b>Total Submissions:</b> 19482</td><td width="10px"></td><td><b>Accepted:</b> 10866</td></table></div><p class="pst">Description</p><div class="ptx" lang="en-US">It is well known that a human gene can be considered as a sequence, consisting of four nucleotides, which are simply denoted by four letters, A, C, G, and T. Biologists have been interested in identifying human genes and  determining  their  functions,  because  these  can  be  used  to  diagnose  human  diseases  and  to  design  new drugs for them. <br> <br>A human gene can be  identified  through a  series of  time-consuming biological experiments, often with  the help of computer programs. Once a sequence of a gene is obtained, the next job is to determine its function. <br>One of the methods for biologists to use in determining the function of a new gene sequence that they have just identified is to search a database with the new gene as a query. The database to be searched stores many gene sequences and their functions – many researchers have been submitting their genes and functions to the database and the database is freely accessible through the Internet. <br> <br>A database  search will  return a  list of gene  sequences  from  the database  that are  similar to  the query gene. <br>Biologists  assume  that  sequence  similarity  often  implies  functional  similarity.  So,  the  function  of  the  new gene might be one of the functions that the genes from the list have. To exactly determine which one is the right one another series of biological experiments will be needed. <br> <br>Your job is to make a program that compares two genes and determines their similarity as explained below. Your program may be used as a part of the database search if you can provide an efficient one.  <br>Given two genes AGTGATG and GTTAG, how similar are they? One of the methods to measure the similarity <br>of two genes is called alignment. In an alignment, spaces are inserted, if necessary, in appropriate positions of <br>the genes to make them equally long and score the resulting genes according to a scoring matrix.  <br> <br>For example, one space is inserted into AGTGATG to result in AGTGAT-G, and three spaces are inserted into GTTAG  to  result  in –GT--TAG. A  space  is denoted by  a minus  sign  (-). The  two genes are now of  equal <br>length. These two strings are aligned:  <br> <br>AGTGAT-G <br>-GT--TAG  <br> <br>In this alignment, there are four matches, namely, G in the second position, T in the third, T in the sixth,  and G in the eighth. Each pair of aligned characters is assigned a score according to the following scoring matrix. <br><center><img src="images/1080/1080_1.gif"></center><br>  denotes  that  a  space-space match  is  not  allowed. The  score  of  the  alignment  above  is  (-3)+5+5+(-2)+(-3)+5+(-3)+5=9. <br> <br>Of course, many other alignments are possible. One is shown below (a different number of spaces are inserted into different positions): <br> <br>AGTGATG <br>-GTTA-G <br> <br>This alignment gives a score of  (-3)+5+5+(-2)+5+(-1) +5=14. So, this one is better than the previous one. As a matter of fact, this one is optimal since no other alignment can have a higher score. So, it is said that the <br>similarity of the two genes is 14. </div><p class="pst">Input</p><div class="ptx" lang="en-US">The input consists of T  test cases. The number of test cases  ) (T  is given in the first line of the input file. Each test case consists of two lines: each line contains an integer, the length of a gene, followed by a gene sequence. The length of each gene sequence is at least one and does not exceed 100. </div><p class="pst">Output</p><div class="ptx" lang="en-US">The output should print the similarity of each test case, one per line. </div><p class="pst">Sample Input</p><pre class="sio">2 
7 AGTGATG 
5 GTTAG 
7 AGCTATT 
9 AGCTTTAAA </pre><p class="pst">Sample Output</p><pre class="sio">14
21 </pre><p class="pst">Source</p><div class="ptx" lang="en-US"><a href="searchproblem?field=source&key=Taejon+2001">Taejon 2001</a></div>